File: internal/service/dto/course.go
package dto

import "time"

type CourseCreateRequest struct {
	CourseName      string  `json:"courseName" binding:"required"`
	Subject         string  `json:"subject" binding:"required"`
	CourseType      string  `json:"courseType" binding:"required"`
	DurationMinutes int32   `json:"durationMinutes"`
	FeeStandard     float64 `json:"feeStandard"`
	Remark          string  `json:"remark"`
	Status          string  `json:"status"`
}

type CourseUpdateRequest map[string]interface{}

type CourseResponse struct {
	ID              int32     `json:"id"`
	CourseName      string    `json:"courseName"`
	Subject         string    `json:"subject"`
	CourseType      string    `json:"courseType"`
	DurationMinutes int32     `json:"durationMinutes"`
	FeeStandard     float64   `json:"feeStandard"`
	Remark          string    `json:"remark"`
	Status          string    `json:"status"`
	CreatedAt       time.Time `json:"createdAt"`
	UpdatedAt       time.Time `json:"updatedAt"`
}

type CourseListResponse []CourseResponse

File: frontend/web/src/api/course.ts
import request from '@/utils/request'
import type { ApiListResult, ApiResponse } from '@/types/api'

export interface CourseItem {
  id: number
  courseName: string
  subject: string
  courseType: string
  durationMinutes: number
  feeStandard: number
  status: string
  remark?: string
  createdAt?: string
  updatedAt?: string
}

export interface CourseFormModel {
  courseName: string
  subject: string
  courseType: string
  durationMinutes: number
  feeStandard: number
  status: string
  remark: string
}

export interface CourseSearchParams {
  courseName?: string
  subject?: string
  status?: string
  pageNum?: number
  pageSize?: number
}

export function fetchCourseList(params?: CourseSearchParams) {
  return request.get<ApiResponse<ApiListResult<CourseItem> | CourseItem[]>>('/courses', { params })
}

export function createCourse(payload: CourseFormModel) {
  return request.post<ApiResponse<{ id: number }>>('/courses', payload)
}

export function updateCourse(id: number | string, payload: CourseFormModel) {
  return request.put<ApiResponse<{ rows_affected: number }>>(`/courses/${id}`, payload)
}

export function deleteCourse(id: number | string) {
  return request.delete<ApiResponse<{ rows_affected: number }>>(`/courses/${id}`)
}
